### PR TITLE
mpvScripts.simple-mpv-webui: init at 1.0.0

### LINKED ISSUE #88584: mpv crashes with scripts using luasocket
**Describe the bug**

When mpv is compiled with samba support, running mpv with a script that uses luasocket will trigger a segmentation fault. This happens because both luasocket and samba export symbols with the same name (such as `socket_listen` and `socket_create`). When the luasocket C code attempts to call its own functions, it will end up calling samba functions. This will obviously crash.

If I override mpv with `{ sambaSupport = false; }` it works ok, but I'm not sure if disabling samba support is desired.

**To Reproduce**
1. Create the following script and save it into `/tmp/script.lua`:
```lua
local socket = require("socket")
local server = socket.bind("0.0.0.0", 44444)
print("bind successful")
```
2. Run `nix-shell -p mpv`
3. Run `mpv /dev/null --script=/tmp/script.lua`

**Expected behavior**

Running the previous command should produce the following output:
```
[script] bind successful
Failed to recognize file format.


Exiting... (Errors when loading file)
```

instead of crashing with a segmentation fault.

**Additional context**

* To test this bug it is required to use a nixpkgs version that has the commit a6dfaad92241ff299131d49d71c53ca4e6661e12. Otherwise, mpv won't be able to import luasocket.
* Samba support for mpv is enabled in Linux systems. Therefore, this issue doesn't exist in darwin systems.

**Notify maintainers**
@AndersonTorres @fpletz @globin @Ma27 @tadeokondrak 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.33, NixOS, 20.09pre222244.22a3bf9fb9e (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- mpv
- luasocket
- samba
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by cript0nauta**
###### Motivation for this change

Adding the simple-mpv-webui mpv plugin and fix #88584

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub }:
+stdenv.mkDerivation {
+  pname = "simple-mpv-ui";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "open-dynaMIX";
+    repo = "simple-mpv-webui";
+    rev = "806d2a1465985ebe54e549ea94fec2f0d0c83dcf";
```
Comment (doronbehar) [inline pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix:9]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub }:
```
Comment (doronbehar) [inline pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix:1]: Personal preference that makes future diffs easier to review & read:

```suggestion
{ stdenv
, fetchFromGitHun
}:
```

File: pkgs/applications/video/mpv/default.nix [hunk 3]
```diff
@@ -39,7 +39,7 @@
 , libpngSupport      ? true,           libpng        ? null
 , pulseSupport       ? config.pulseaudio or stdenv.isLinux, libpulseaudio ? null
 , rubberbandSupport  ? stdenv.isLinux, rubberband    ? null
-, sambaSupport       ? stdenv.isLinux, samba         ? null
+, sambaSupport       ? false,          samba         ? null
```
Comment (doronbehar) [inline pkgs/applications/video/mpv/default.nix:42]:
```suggestion
# NOTE: samba support should be removed on the next mpv release, see also:
# https://github.com/NixOS/nixpkgs/pull/89145#issuecomment-636424362
# Please remove this line on the next mpv release.
, sambaSupport       ? false,          samba         ? null
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Comment by doronbehar** [comment]
Also, if instead of:

> fixing #88584

You'd use:

> fix #88584

That issue will be closed automatically upon merge of this PR.

File: pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix (hunk 2 above)
Comment (cript0nauta) [reply pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix:1]: I used nixfmt to format this expressions, but sure I can change it!

**Comment by cript0nauta** [comment]
> Also, if instead of:
> 
> > fixing #88584
> 
> You'd use:
> 
> > fix #88584
> 
> That issue will be closed automatically upon merge of this PR.

I put `Closes #88584` in the commit message so this should close the issue once this is merged.

**Comment by doronbehar** [comment]
> I put Closes #88584 in the commit message so this should close the issue once this is merged.

Hmm, I guess you are right but GitHub doesn't pick this to the "Linked issues" section at the right bar of the PR...

File: pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix (hunk 1 above)
Comment (cript0nauta) [reply pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix]: Applied and amended. Thanks for the suggestion!

File: pkgs/applications/video/mpv/default.nix (hunk 3 above)
Comment (cript0nauta) [reply pkgs/applications/video/mpv/default.nix:45]: Applied and amended. Thanks for the suggestion!

File: pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix (hunk 2 above)
Comment (cript0nauta) [reply pkgs/applications/video/mpv/scripts/simple-mpv-webui.nix]: Applied and amended. Thanks for the suggestion!

**Comment by cript0nauta** [comment]
> Hmm, I guess you are right but GitHub doesn't pick this to the "Linked issues" section at the right bar of the PR...

Oh ok, I updated the description now as you suggested.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -20967,6 +20967,7 @@ in
   mpvScripts = recurseIntoAttrs {
     convert = callPackage ../applications/video/mpv/scripts/convert.nix {};
     mpris = callPackage ../applications/video/mpv/scripts/mpris.nix {};
+    simple-mpv-webui = callPackage ../applications/video/mpv/scripts/simple-mpv-webui.nix {};
```
Comment (doronbehar) [inline pkgs/top-level/all-packages.nix:20970]: Perhaps:

```suggestion
    simple-webui = callPackage ../applications/video/mpv/scripts/simple-mpv-webui.nix {};
```

Or even:

```suggestion
    webui = callPackage ../applications/video/mpv/scripts/simple-mpv-webui.nix {};
```

It's up to your decision. The last option seems more concise :thinking: ... The reasoning is that `mpv` is already in the attribute path of this derivation.

**Review by doronbehar — APPROVED** [review]
It works and it's pretty cool. Thanks!

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (cript0nauta) [reply pkgs/top-level/all-packages.nix:20970]: I personally prefer to use the same name as in the GitHub repo in order to make searching easier, even if it implies repeating "mpv". I think this approach is used in other places, such as Python packages (e.g. `pythonPackages.python-dateutil`).

**Pull request merged**